### ISSUE TITLE
Design Doc Review #1 - MVP List/Schema/Backend Routes

### ISSUE THREAD

**Issue by mmcnulty20** [waj919/ThisCord#1]
I've reviewed your first three docs! For comments phrased as a question, you can either answer via a comment here or we can discuss via Slack if it is a choice you are making or a point you can explain, or you can modify the docs to resolve it.

# Wiki Page Home

- [x] Is the first page you see upon entering the wiki
- [x] Contains a welcome message
- [x] Contains a link/placeholder for a link to the live page
- [x] All links in the right sidebar should contain each wiki page and link to the correct page
- [x] Correctly formatted
    - [x] each wiki page is listed in bullet points
    - [x] all links route the correct page
    
### Comments

# MVP List

- [x] Should have 7 MVPs.
    - [x] 3 of those are User Auth, Heroku, and Production README.
    - [x] The other 4 are from the MVP List or they have clarified them with you
- [x] Contains a description sentence of the app
- [x] Includes two to three detailed bullets on functionality and presentation of feature
- [x] At least one CRUD feature, which states what CRUD operations are planned (creation, reading, updating, deletion)
- [x] Estimates how long it will take the code each MVP
- [x] Correctly formatted
    - [x] MVPs are listed in an ordered list
    - [x] Each MVP is broken down into bullet points
    
### Comments

- [x] Mostly good but add a bit more detail/clarity for the MVP bullets. Examples of things to clarify: What does editing server settings mean? Can any user delete a server? A channel? Can users simply create channels anywhere? Users don't have multiple channels in a server (which implies different users may have different channels in a given server) but instead servers have channels, mention websockets in livechat, either omit "via servers" or specify "via private servers" for #6, etc 
- [x] Don't mention video in the description sentence if you don't intend to implement that feature.
- [x] "Users need to be signed in to have access channels" - users need to be signed in for more than that
- [x] Users can sign up as well as log in/out
- [x] Not required, but I'd recommend thinking of a couple bonus features you may want to implement if you get to them or on the job search! 

# Database Schema

- [x] Contains correct datatypes
- [x]  Contains appropriate constraints/details
    - [x]  primary key
    - [x]  not null
    - [x] unique
    - [x] indexed
    - [x] foreign key
- [x] Contains bullet points after the table that state which foreign keys will reference to which table, or references to the associations which will be made
    - [x] foreign key and table name are lowercased, snake_cased and `back_ticked`
- [x] Correctly formatted
    - [x] schema is written in a table format
    - [x] the table's name are lowercased, snake_cased and `back_ticked`
    - [x] the table header column names are bolded
    - [x] columns names are lowercased and snaked_cased and `back_ticked`
    
### Comments

- [x] Make sure all indexes and unique constraints are both on the details column and explained in bullets beneath 
- [x] Servers don't have an `owner_id` to be indexed
- [x] Do you intend to be able to search for servers by name?
- [x] What would a server type look like? When would it be null? What do you mean by 'template'?
- [x] The "chat" a `chat_id` references is not a table name (also ensure these are bullet points)
- [x] I would recommend looking into polymorphic associations with messages and having a "direct message" be a messageable entity rather than having DMs individually hold that information (this also would allow easier creation of multi-person DMs if you wanted to implement that)
- [x] Where would we store the servers a user has joined?

# Backend Routes

- [x]  Contains the following sections: HTML, API Endpoints(Backend)
- [x]  Each route has a description
- [x]  API Endpoint routes contains wildcard variables written in `snake_case`
- [x]  Routes does not contain superfluous routes
- [x]  Have API routes that will allow the front end to get all info it needs and does not have unneeded routes:
    -  probably doesn't need a `GET likes` api endpoint because that info comes through the post show
    
### Comments

- [x] When would you use the `GET /api/users/:id` route?
- [x] Clarify the information you will return from POST and UPDATE requests
- [x] When would you want to get all servers from the database? Also clarify what info about servers would be returned.
- [x] What is "all its information" for the server show route?
- [x] How will a user join a server? How will you handle DMs?

**Comment by waj919** [comment]
User will be able to join a server with an invite code. DMs will have a server ID that is specifically for messaging people directly

**Comment by mmcnulty20** [comment]
Invite codes should be bonus functionality if anything. Instead you should be able to search for and join a server as part of the servers MVP. The questions here are primarily because you currently don't seem to have the infrastructure to join servers or to DM users.

When you say a DM will have a server ID, can you elaborate on what that would mean? Let's say I've got private messages with you and with Paloma while on ThisCord. What would sending a message to each of you separately look like to the backend? Is it different for the first time to send a message to a user and subsequent times? How does that work? Can you private message multiple users? How is that handled? Does a DM have a name and owner_id? How do you determine the owner of a DM? Etc

**Comment by waj919** [comment]
Ok I see,  I will have to implement a search function to allow users to join servers without invite codes

DMs would be in their own "server" where all your messages with friends would basically be a text channel with that specific person. If you were to send a message to each of us, our messages would basically be a channel. The database would find the DM server and then the "DM" (channel) you would want to message.  You would be able to private message as many people as you want

**Comment by mmcnulty20** [comment]
Okay! In which case rather than a DM having a `server_id`, it sounds like a DM is just a server entity (perhaps with a particular name indicating such) unless I'm misunderstanding your intended implementation!

**Comment by waj919** [comment]
Yeah! The actual discord website has dms under the route `/channels/@me`. so I was going to do the same thing
